exp/api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -116,6 +116,10 @@ type AzureManagedControlPlaneSpec struct {
 	// APIServerAccessProfile is the access profile for AKS API server.
 	// +optional
 	APIServerAccessProfile *APIServerAccessProfile `json:"apiServerAccessProfile,omitempty"`
+
+	// AzureAvailabilityStatusEnabled determines whether or not the AzureResourceAvailable status condition will be reported.
+	// +optional
+	AzureAvailabilityStatusEnabled bool `json:"azureAvailabilityStatusEnabled"`
```

Comment:
+1 to the suggestion to add a toggle so this feature can be enabled by users who understand it for now until latency is fixed

However, we need to be very careful with adding API fields to implement temporary toggles like this: once it's in the API spec, we can't remove it without making a breaking change which requires a new API version. How about instead using a feature flag at the controller level, similar to the AKS feature flag itself?

The only downside there is that you would enable it for all your clusters at once, not per cluster. I don't know if that's really a downside though, since I can't really see a use case for someone wanting to enable resource health for some clusters but not others.

Reply:
Let's bring this up in office hours this week and solicit feedback from the user community about how to implement this. I can definitely see a reason why a user would prefer their own "resource health" instrumentation (perhaps there are other vectors for latency given how much metrics tooling has the opportunity to touch every network packet). And so this would be a case where including it in the API does make sense (it would be a durable configuration interface).

But let's ask and see what users think.

cc @mtougeron @zmalik @sonasingh46 @luthermonson

Reply:
> I can definitely see a reason why a user would prefer their own "resource health" instrumentation

I don't fully understand this comment, why would one prevent you from doing the other? The `Status` field is just extra information, once AKS fixes the initial 10 minute initial API response of `Unknown` this would be useful information for any user, regardless of network stack or metric tooling as far as I can tell.

IMO a feature flag would be a better fit, in this case, at least to start, and then we can decide to move it into the API later if we see a need. Doing the reverse would be much harder and breaking. If we do want to move it into a durable API user exposed Spec field, we also want to do this with a durable API design, potentially add a new struct so we don't later end up with a flat list of toggles in the CRD `Spec`.

Reply:
@mtougeron @zmalik @sonasingh46 @luthermonson Any thoughts on the above? The tl;dr so far is:
- Azure has a generic "Resource Health" service that AKS hooks into to report certain cluster issues via the Azure API.
- The goal of this PR is to report this same info in the `status` of the AzureManagedControlPlane.
- There's a quirk in Azure or AKS where a harmless (as far as we know) error gets reported for the first ~20 minutes after a cluster is fully provisioned, blocking the `Ready` condition on the AzureManagedControlPlane for that period of time. When we mentioned the delay issue to AKS, they thought 20 minutes is definitely worth looking into improving, but I got the sense that it would still likely end up being some single-digit number of minutes at best.

So the questions we're looking to answer are:
- Is this feature something we need to have a way to disable?
- If so, how should it be disabled? The two options presented so far are via the API in something like the AzureManagedControlPlane's `spec.azureAvailabilityStatusEnabled` or via a feature flag.

Reply:
Sorry, I missed this the other week when I was first flagged...

I think I agree with @CecileRobertMichon where this should be a feature flag rather than an API field. I don't think it is something that should be set on the cluster level to override what Azure is telling CAPZ. I think it's more appropriate to tell CAPZ how to handle the response/status from Azure. IMO, it's more of a change of application behavior than a change of cluster definition.

Reply:
sorry, I also missed this 
This is actually great as we would like to see it in the CR. And toggle feature is also a good way. The reasonings is that Azure Resource Health also counts in for the API requests which adds up to the total limit of requests. We are trying to reduce that as we end up hitting subscription limits in some cases. Having that toggle available would be helpful. 

As feature flag or inside spec, I don't have strong opinions. But would be really helpful as having a separate path such as Azure Metrics Exporter would allow us to keep this disabled. To not to do same query twice.

Reply:
+1 to the feature flag.

I suppose the right thing to do is to turn this off by default. @CecileRobertMichon do we have an existing pattern where we would introduce a feature flag default-enabled?

Reply:
we don't have an existing one that on by default but the way you'd do that it to set the environment variable default value to true in manager.yaml https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/config/manager/manager.yaml#L26

Reply:
Sounds good, I'll plan to add the feature flag back. I think that would also give us a path toward moving the toggle into the API if we find that should be a more permanent option.

Is it worth still explicitly ignoring this health metrics error we see for every cluster created even when the feature is enabled?

Reply:
+1